core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -75,7 +75,7 @@ private[sql] object PartitionedPhysicalRDD {
   }
 }
 
-trait PartitionedDataSourceScan extends PrunedFilteredScan {
+trait PartitionedDataSourceScan extends PrunedUnsafeFilteredScan {
```

Comment:
With these changes, the explicit UnsafeProjection in PartitionedPhysicalRDD.doExecute is unnecessary and should be removed. Without that removal you will not get the expected boost at all. For TPCH q1 that change is what helped most (~30% improvement) while these changes by themselves had minimal impact.